esdt-safe/interactor/src/interactor_main.rs

```diff
@@ -131,66 +153,74 @@ struct ContractInteract {
     alice_address: Address,
     mike_address: Address,
     judy_address: Address,
-    contract_code: BytesValue,
-    fee_market_code: BytesValue,
-    price_aggregator_code: BytesValue,
-    header_verifier_code: BytesValue,
+    esdt_safe_code: String,
+    fee_market_code: String,
+    price_aggregator_code: String,
+    header_verifier_code: String,
+    testing_sc_code: String,
     state: State,
 }
 
 impl ContractInteract {
     async fn new() -> Self {
         let mut interactor = Interactor::new(GATEWAY, false).await;
-        let wallet_address = interactor.register_wallet(test_wallets::frank()).await;
-        let bob_address = interactor.register_wallet(test_wallets::bob()).await;
+
+        interactor.set_current_dir_from_workspace("esdt-safe/interactor");
+
+        let wallet_address = interactor.register_wallet(test_wallets::bob()).await;
+        let frank_address = interactor.register_wallet(test_wallets::frank()).await;
         let alice_address = interactor.register_wallet(test_wallets::alice()).await;
         let mike_address = interactor.register_wallet(test_wallets::mike()).await;
         let judy_address = interactor.register_wallet(test_wallets::judy()).await;
 
-        let contract_code = BytesValue::interpret_from(
-            "mxsc:../output/esdt-safe.mxsc.json",
-            &InterpreterContext::default(),
-        );
+        let current_dir = find_current_workspace().unwrap();
+        println!("Current directory is: {}", current_dir.display());
+
+        let repo_dir = current_dir
+            .ancestors()
+            .nth(2)
+            .expect("Failed to go up 2 levels");
+        println!("Repo directory is: {}", repo_dir.display());
 
-        let fee_market_code = BytesValue::interpret_from(
-            "mxsc:contract-codes/fee-market.mxsc.json",
-            &InterpreterContext::default(),
-        );
+        let fee_market_code = "../../fee-market/output/fee-market.mxsc.json".to_owned();
 
-        let price_aggregator_code = BytesValue::interpret_from(
-            "mxsc:contract-codes/multiversx-price-aggregator-sc.mxsc.json",
-            &InterpreterContext::default(),
-        );
+        let header_verifier_code =
+            "../../header-verifier/output/header-verifier.mxsc.json".to_owned();
 
-        let header_verifier_code = BytesValue::interpret_from(
-            "mxsc:contract-codes/header-verifier.mxsc.json",
-            &InterpreterContext::default(),
-        );
+        let esdt_safe_code = "../output/esdt-safe.mxsc.json".to_owned();
+
+        let price_aggregator_code =
+            "contract-codes/multiversx-price-aggregator-sc.mxsc.json".to_owned();
+
+        let testing_sc_code = "../../testing-sc/output/testing-sc.mxsc.json".to_owned();
 
         ContractInteract {
             interactor,
             wallet_address,
-            bob_address,
+            bob_address: frank_address,
```

Comment:
Aici nu poti lasa direct frank_address? Altfel e putin confusing

Reply:
I forgot to modify it since bob's address was out of funds at some point